### PR TITLE
langchain_google_vertexai : added logic to override get_num_tokens_from_messages() for ChatVertexAI

### PR DESCRIPTION
**Pull request by Adi8885**
<!-- Thank you for contributing to LangChain!

Replace this entire comment with:
  - **Description: added logic to override get_num_tokens_from_messages() for ChatVertexAI. Currently ChatVertexAI was inheriting get_num_tokens_from_messages() from BaseChatModel which in-turn was calling GPT-2 tokenizer
  - **Issue: NA
  - **Dependencies: NA
  - **Twitter handle:@aditya_rane

@lkuligin for review

### CODE REVIEW

File: libs/partners/google-vertexai/tests/integration_tests/test_chat_models.py
Comment (efriis) [inline libs/partners/google-vertexai/tests/integration_tests/test_chat_models.py:1]: why remove these two tests?

Comment (Adi8885) [reply libs/partners/google-vertexai/tests/integration_tests/test_chat_models.py:1]: my bad , it was causing some issues on my local setup. Added them back

**Comment by turanbulmus** [comment]
@efriis can we merge this as soon as possible? We have a production case depending on this PR. Thanks a lot for your support

**Comment by Adi8885** [comment]
@lkuligin For review

**Comment by thomasmenard-oa** [comment]
@efriis @lkuligin we need this PR merged and pushed into a new version langchain_google_vertexai.
We have production deployement awaiting for this fix.
Thanks :)

**Comment by Adi8885** [comment]
@efriis @lkuligin Kindly let me know if there are any blockers for this PR , we have critical deployment awaiting this change?

**Comment by thomasmenard-oa** [comment]
Looping @hwchase17. Sorry to ping you on that one but we need this PR merged please

File: libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py [hunk 1]
```diff
@@ -531,3 +531,26 @@ def _start_chat(
             )
         else:
             return self.client.start_chat(message_history=history.history, **kwargs)
+
+    def get_num_tokens_from_messages(self, messages: List[BaseMessage]) -> int:
```
Comment (lkuligin) [inline libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py:535]: it's probably better to move the `get_num_tokens` method from `VertexAI` to `_VertexAICommon`, and then everything should work just fine.

**Review by lkuligin — CHANGES_REQUESTED** [review]

**Comment by lkuligin** [comment]
also, it might be a good idea to add an integration test that uses `ConversationMemoryBuffer` since this is what we're trying to fix here if I'm not wrong.

File: libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py (hunk 1 above)
Comment (Adi8885) [reply libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py]: Done

**Comment by Adi8885** [comment]
> also, it might be a good idea to add an integration test that uses `ConversationMemoryBuffer` since this is what we're trying to fix here if I'm not wrong.

@lkuligin Were testing "ConversationSummaryBufferMemory" , but not sure what we want to check integration test. I've checked locally , it doesn't re-direct to tokenizer download

File: libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py (hunk 1 above)
Comment (lkuligin) [reply libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py]: we don't even need `get_num_tokens_from_messages`, we need to move an existing `get_num_tokens` method and then it should work:
https://github.com/langchain-ai/langchain/blob/f027696b5f068bacad96a9356ae196f5de240007/libs/core/langchain_core/language_models/base.py#L275

Comment (Adi8885) [reply libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py]: @lkuligin This approach would result in multiple calls to the count_token endpoint i.e separately for each message. This approach in this PR collates all text in messages and calls the  count_token endpoint only once. 
```
https://github.com/langchain-ai/langchain/blob/f027696b5f068bacad96a9356ae196f5de240007/libs/core/langchain_core/language_models/base.py#L275
```

Comment (lkuligin) [reply libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py]: fair enough, but let's still move the `get_num_tokens`, please. 
and I'm not sure that difference (1 large call vs n smaller calls) is significant.

@sasha-gitg any thoughts on this?

Comment (lkuligin) [reply libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py]: additional thoughts: 
1. let's move `get_num_tokens` pls
2. you can always collapse message together:
either with
```
from vertexai.generative_models._generative_models import _PreviewGenerativeModel

get_buffer_string(
    messages, 
    human_prefix=_PreviewGenerativeModel._USER_ROLE,
    ai_prefix=_PreviewGenerativeModel._MODEL_ROLE)
```
or with 
```
self.client..count_tokens([m.content for m in messages])
```

**Review by lkuligin — APPROVED** [review]

**Review by efriis — APPROVED** [review]

**Pull request merged**